src/Chainweb/Mempool/InMem.hs

```diff
@@ -150,18 +183,26 @@ markValidatedInMem lock txs = withMVarMasked lock $ \mdata -> do
     let pref = _inmemPending mdata
     modifyIORef' pref $ \psq -> foldl' (flip PSQ.delete) psq txs
 
-
 ------------------------------------------------------------------------------
 insertInMem :: InMemConfig t    -- ^ in-memory config
             -> MVar (InMemoryMempoolData t)  -- ^ in-memory state
             -> Vector t  -- ^ new transactions
             -> IO ()
 insertInMem cfg lock txs0 = do
     txs <- V.filterM preGossipCheck txs0
-    withMVarMasked lock $ \mdata -> do
+    i <- withMVarMasked lock $ \mdata -> do
         newHashes <- V.mapM (insOne mdata) txs
+        modifyIORef' (_inmemInserted mdata) (+ (length txs))
         modifyIORef' (_inmemRecentLog mdata) $
             recordRecentTransactions maxRecent newHashes
+        readIORef (_inmemInserted mdata)
+
+    unless (i < 5000) $ do
+        withMVarMasked lock $ \mdata -> do
+            modifyIORef' (_inmemPending mdata) $ \ps -> do
+                if (PSQ.size ps > 10000)
```

Comment:
Don't you have to update `_inmemInserted` here?

I think if you're going to do this, it might be best just to drop the new `_inmemInserted` field and just always do the check. If we are bounding the size of the set then calling `PSQ.size` every time is not that expensive.

Reply:
yes, `_inmemInserted` should be reset here.